controllers/metal3.io/host_state_machine.go

```diff
@@ -227,6 +235,37 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	return true
 }
 
+func (hsm *hostStateMachine) checkDetachedHost(info *reconcileInfo) (result actionResult) {
+	// If the detached annotation is set we remove any host from the
+	// provisioner and take no further action
+	// Note this doesn't change the current state, only the OperationalStatus
+	annotations := hsm.Host.GetAnnotations()
+	if annotations != nil {
+		if _, ok := annotations[metal3v1alpha1.DetachedAnnotation]; ok {
+			// Only allow detaching hosts in Provisioned/ExternallyProvisioned states
+			if info.host.Status.Provisioning.State == metal3v1alpha1.StateProvisioned || info.host.Status.Provisioning.State == metal3v1alpha1.StateExternallyProvisioned {
+				return hsm.Reconciler.actionDetached(hsm.Provisioner, info)
+			}
+		}
+	}
+	if info.host.Status.ErrorType == metal3v1alpha1.DetachError {
```

Comment:
So, this is the case where the "detached" annotation was present, the detach operation resulted in a DetachError but then the annotation was removed? If that is the case,  is there a need to also set the OperationalStatus to some thing like metal3v1alpha1.OperationalStatusOK?

Reply:
I just happened to be reading the source for `clearError()` and it does do that.

Reply:
Yeah that's already handled by `clearError()`